files/image_config/hostcfgd/hostcfgd

```diff
@@ -42,11 +42,12 @@ def obfuscate(data):
 
 
 def update_feature_state(feature_name, state):
+    feature_suffix = "timer" if feature_name in ["snmp", "telemetry"] else "service"
     if state == "enabled":
         start_cmds = []
-        start_cmds.append("sudo systemctl unmask {}.service".format(feature_name))
-        start_cmds.append("sudo systemctl enable {}.service".format(feature_name))
-        start_cmds.append("sudo systemctl start {}.service".format(feature_name))
+        start_cmds.append("sudo systemctl unmask {}.{}".format(feature_name, feature_suffix))
+        start_cmds.append("sudo systemctl enable {}.{}".format(feature_name, feature_suffix))
+        start_cmds.append("sudo systemctl start {}.{}".format(feature_name, feature_suffix))
```

Comment:
Wondering if we can simplify this by removing the suffix altogether to avoid the need to maintain the list of timers. Can you verify whether systemctl is smart enough to use the timer file, if present, rather than the service file if we omit the suffix altogether?

Reply:
The notes section in the documentation stated that .timer unit should be enabled

https://wiki.archlinux.org/index.php/Systemd/Timers

Reply:
@jleveque I tried this [image](https://sonic-jenkins.westus2.cloudapp.azure.com/job/broadcom/job/buildimage-brcm-all-released-pr/577/artifact/target/sonic-broadcom.bin) that does not have suffixes after service name. It doesn't see to work:

```
admin@str-s6000-acs-14:~$ docker ps -a
CONTAINER ID        IMAGE                                COMMAND                  CREATED              STATUS              PORTS               NAMES
18bafe7ce51b        docker-snmp-sv2:latest               "/usr/bin/supervisord"   11 seconds ago       Up 7 seconds                            snmp
e2a9e3fc7be9        docker-sonic-telemetry:latest        "/usr/bin/supervisord"   21 seconds ago       Up 19 seconds                           telemetry
5af1b804930c        docker-sonic-mgmt-framework:latest   "/usr/bin/supervisord"   36 seconds ago       Up 33 seconds                           mgmt-framework
27db04f8b5a6        docker-lldp-sv2:latest               "/usr/bin/docker-lld…"   37 seconds ago       Up 33 seconds                           lldp
cce6d141dc36        docker-router-advertiser:latest      "/usr/bin/docker-ini…"   37 seconds ago       Up 34 seconds                           radv
1f0921cc2ff9        docker-dhcp-relay:latest             "/usr/bin/docker_ini…"   38 seconds ago       Up 34 seconds                           dhcp_relay
48a86ed6dfdc        docker-syncd-brcm:latest             "/usr/bin/supervisord"   48 seconds ago       Up 46 seconds                           syncd
ae83b206baf7        docker-teamd:latest                  "/usr/bin/supervisord"   50 seconds ago       Up 49 seconds                           teamd
eb511fe3334d        docker-orchagent:latest              "/usr/bin/docker-ini…"   About a minute ago   Up About a minute                       swss
0ec9f91d7714        docker-platform-monitor:latest       "/usr/bin/docker_ini…"   About a minute ago   Up About a minute                       pmon
343839fddaac        docker-fpm-frr:latest                "/usr/bin/supervisord"   About a minute ago   Up About a minute                       bgp
deb8fa973725        docker-database:latest               "/usr/local/bin/dock…"   About a minute ago   Up About a minute                       database
```

It looks as if both .service and .timer systemd units were started at the same time which defeat the delay timer purpose:

```
Aug 19 17:25:48.019367 str-s6000-acs-14 INFO hostcfgd: Running cmd: 'sudo systemctl unmask snmp'
Aug 19 17:25:48.713173 str-s6000-acs-14 INFO hostcfgd: Running cmd: 'sudo systemctl enable snmp'
Aug 19 17:25:49.452243 str-s6000-acs-14 INFO hostcfgd: Running cmd: 'sudo systemctl start snmp'
Aug 19 17:25:49.574206 str-s6000-acs-14 INFO systemd[1]: Stopped Delays snmp container until SONiC has started.
Aug 19 17:25:54.406954 str-s6000-acs-14 INFO snmp.sh[3558]: Creating new snmp container with HWSKU Force10-S6000
Aug 19 17:25:54.961290 str-s6000-acs-14 INFO snmp.sh[3558]: 18bafe7ce51b1da7a6a76057f0ba293dc8aecf8c8663af7d51e0a07c1058414f
Aug 19 17:25:56.872962 str-s6000-acs-14 INFO snmp.sh[3558]: 1
Aug 19 17:25:58.244876 str-s6000-acs-14 INFO snmp.sh[3558]: snmp
Aug 19 17:25:58.276946 str-s6000-acs-14 INFO hostcfgd: Feature 'snmp' is enabled and started
Aug 19 17:26:19.356662 str-s6000-acs-14 INFO snmp.sh[4115]: snmp
Aug 19 17:26:19.358465 str-s6000-acs-14 INFO snmp.sh[3768]: 137
Aug 19 17:26:19.384640 str-s6000-acs-14 INFO systemd[1]: Started Delays snmp container until SONiC has started.
Aug 19 17:26:19.389841 str-s6000-acs-14 INFO systemd[1]: Stopped Delays snmp container until SONiC has started.
Aug 19 17:26:23.746481 str-s6000-acs-14 INFO snmp.sh[4339]: Starting existing snmp container with HWSKU Force10-S6000
Aug 19 17:26:25.469834 str-s6000-acs-14 INFO snmp.sh[4339]: 0
Aug 19 17:26:26.707544 str-s6000-acs-14 INFO snmp.sh[4339]: snmp
Aug 19 17:26:38.007094 str-s6000-acs-14 INFO snmp#liblogging-stdlog:  [origin software="rsyslogd" swVersion="8.24.0" x-pid="17" x-info="http://www.rsyslog.com"] start
```

Reply:
Bummer. I'd still like to prevent the need to keep track of which services have a timer. Maybe we could add a function to check if a timer file is present for the service, so it can be done dynamically?